zebra-network/src/address_book.rs

```diff
@@ -104,6 +139,14 @@ impl AddressBook {
             recent_peers = self.recently_live_peers().count(),
         );
 
+        // Drop any unspecified or client addresses.
+        //
+        // Communication with these addresses can be monitored via Zebra's
+        // metrics. (The address book is for valid peer addresses.)
+        if !new.is_valid_for_outbound() {
+            return;
+        }
+
```

Comment:
If the node we've just connected to doesn't have the node service bit, also remove that address from our address book.

Reply:
Fixed in 41822b6f244b66997d82f9b9072c577b9dc16809